src/tensorflow/tensorflow-rnnlm-lib.h

```diff
@@ -0,0 +1,156 @@
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_LM_TENSORFLOW_LIB_H_
+#define KALDI_LM_TENSORFLOW_LIB_H_
+
+#include <string>
+#include <vector>
+#include "util/stl-utils.h"
+#include "base/kaldi-common.h"
+#include "fstext/deterministic-fst.h"
+#include "util/common-utils.h"
+#include "tensorflow/core/public/session.h"
+
+using tensorflow::Session;
+using tensorflow::Tensor;
+
+namespace kaldi {
+namespace tf_rnnlm {
+
+struct KaldiTfRnnlmWrapperOpts {
+  std::string unk_symbol;
+  std::string eos_symbol;
+  int32 num_jobs;  // 0 means unlimited
+
+  KaldiTfRnnlmWrapperOpts() : unk_symbol("<oos>"), eos_symbol("</s>"),
+                              num_jobs(1) {}
+
+  void Register(OptionsItf *opts) {
+    opts->Register("unk-symbol", &unk_symbol, "Symbol for out-of-vocabulary "
+                   "words in rnnlm.");
+    opts->Register("eos-symbol", &eos_symbol, "End of setence symbol in "
+                   "rnnlm.");
+    opts->Register("num-jobs", &num_jobs, "Number of jobs for TF computation; "
+                   "0 means unlimited. (default = 1)");
+  }
+};
+
+class KaldiTfRnnlmWrapper {
+ public:
+  KaldiTfRnnlmWrapper(const KaldiTfRnnlmWrapperOpts &opts,
+                      const std::string &rnn_wordlist,
+                      const std::string &word_symbol_table_rxfilename,
+                      const std::string &unk_prob_file,
+                      const std::string &tf_model_path);
+
+  ~KaldiTfRnnlmWrapper() {
+    session_->Close();
+  }
+
+  int32 GetEos() const { return eos_; }
+
+  // get an all-zero Tensor of the size that matches the hidden state of the TF model
+  const Tensor& GetInitialContext() const;
+
+  // get the 2nd-to-last layer of RNN when feeding input of
+  // (initial-context, sentence-boundary)
+  const Tensor& GetInitialCell() const;
+
+  // compute p(word | wseq) and return the log of that
+  // the computation used the input cell,
+  // which is the 2nd-to-last layer of the RNNLM associated with history wseq;
+  //
+  // and we generate (context_out, new_cell) by passing (context_in, word) into the model
+  // if the last 2 pointers are NULL we don't query then in TF session
+  BaseFloat GetLogProb(int32 word, // need the FST word label for computing OOS cost
+                       int32 fst_word,
+                       const Tensor &context_in,  // context to pass into RNN
+                       const Tensor &cell_in,     // 2nd-to-last layer
+                       Tensor *context_out,
+                       Tensor *new_cell);
+
+  // since usually we have a smaller vocab in RNN than the whole vocab,
```

Comment:
need more explanation about this variable, e.g. are the RNN labels  zero based?  What about FST labels that are OOS?  You need to explain what OOS means, too, when uyou use the term.